sources/packages/web/src/components/institutions/modals/EditInstitutionUserModal.vue

```diff
@@ -0,0 +1,264 @@
+                  </span>
+                </v-col>
+              </v-row>
+              <v-row v-for="location in institutionLocationList" :key="location"
+                ><v-col>
+                  <span>{{ location?.name }} </span><br />
+                  <span>
+                    <span>{{ location?.data?.address?.addressLine1 }}, </span>
+                    <span>{{ location?.data?.address?.city }}, </span>
+                    <span>{{ location?.data?.address?.province }}</span>
+                    <br />
+                  </span>
+                </v-col>
+                <v-col>
+                  <Dropdown
+                    v-model="location.userType"
+                    :options="userType"
+                    optionLabel="name"
+                    placeholder="Select a User Role"
+                    :style="{ width: '25vw' }"
+                    :class="invalidUserType ? 'p-invalid' : ''"
+                  />
+                </v-col>
+              </v-row>
+            </span>
+          </span>
+        </form>
+      </v-container>
+    </v-sheet>
+    <template #footer>
+      <v-btn color="primary" outlined @click="closeEditUser()"> Cancel </v-btn>
+      <v-btn color="primary" @click="submitEditUser()">
+        Save Changes
+        <v-icon right>mdi-account </v-icon>
+      </v-btn>
+    </template>
+  </Dialog>
+</template>
+
+<script lang="ts">
+/* eslint-disable */
+import { ref, watch } from "vue";
+import { InstitutionService } from "@/services/InstitutionService";
+import Dialog from "primevue/dialog";
+import Dropdown from "primevue/dropdown";
+import InputSwitch from "primevue/inputswitch";
+import { useToast } from "primevue/usetoast";
+import {
+  InstitutionLocationUserAuthDto,
+  InstitutionUserWithUserType,
+  InstitutionUserRoleLocation,
+  InstitutionUserAuthDetails,
+} from "@/types";
+
+export default {
+  components: { Dialog, Dropdown, InputSwitch },
+  props: {
+    showEditUser: {
+      type: Boolean,
+      default: false,
+    },
+    userType: {
+      type: Array,
+      default: [],
+    },
+    institutionUserName: {
+      type: String,
+      default: "",
+    },
+  },
+  emits: ["updateShowEditInstitutionModal", "getAllInstitutionUsers"],
+  setup(props: any, context: any) {
+    const userData = ref({} as InstitutionLocationUserAuthDto);
+    const isAdmin = ref(false);
+    const toast = useToast();
+    const invalidUserType = ref(false);
+    const display = ref(true);
+    const institutionLocationList = ref();
+    const payLoad = ref({} as InstitutionUserAuthDetails);
+    const closeEditUser = async () => {
+      context.emit("updateShowEditInstitutionModal");
+      invalidUserType.value = false;
+      await getInstitutionUserDetails();
+      checkIsAdmin();
+      await getAndSetAllLocations();
+    };
+    const checkIsAdmin = () => {
+      isAdmin.value =
+        userData.value?.authorizations?.filter((el: any) => el?.authType?.type == "admin")
```

Comment:
The expression could be simplified with the use of "some" instead of "filter" as below.
IMHO this should be resolved by API or some centralized logic. Don't need to change it right the way.

```
userData.value?.authorizations?.some(el => el?.authType?.type === "admin")
```